uuid.go

```diff
@@ -15,6 +16,16 @@ func GenerateRandomBytes(size int) ([]byte, error) {
 	return buf, nil
 }
 
+// GenerateRandomBytesByReader is used to generate random bytes of given size by given Reader.
+func GenerateRandomBytesByReader(size int, reader io.Reader) ([]byte, error) {
+	buf := make([]byte, size)
+	if _, err := reader.Read(buf); err != nil {
```

Comment:
Are we guaranteed that with a nil error io.Reader returns all `size` bytes?  The godoc is a little ambiguous I feel, but see e.g.

> If some data is available but not len(p) bytes, Read conventionally returns what is available instead of waiting for more

Reply:
Well the example in crypto/rand does it this way, so I guess it's okay.

Reply:
good point - this isn't a concern with crypto/rand's Reader because it is using io.ReadFull under the hood which has this behaviour:

`ReadFull reads exactly len(buf) bytes from r into buf. It returns the number of bytes copied and an error if fewer bytes were read. The error is EOF only if no bytes were read. If an EOF happens after reading some but not all the bytes, ReadFull returns ErrUnexpectedEOF. On return, n == len(buf) if and only if err == nil. If r returns an error having read at least len(buf) bytes, the error is dropped.`

I think it makes sense to conserve this behaviour by using ReadFull here